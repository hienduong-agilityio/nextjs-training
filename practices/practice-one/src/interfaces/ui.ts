export interface IInputGroupProps {
  placeholder?: string;
  value?: string;
  buttonText?: string;
  inputName?: string;
  isDisabled?: boolean;
  isLoading?: boolean;
  startIcon?: React.ReactNode;
  customClass?: {
    container?: string;
    inputContainer?: string;
    input?: string;
    button?: string;
  };
  onSubmit?: (event: React.FormEvent<HTMLFormElement>) => void;
}
