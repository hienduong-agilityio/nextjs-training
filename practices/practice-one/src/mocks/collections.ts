import { Category } from '@/types/category';

type HotDeal = {
  name: Category | 'all';
  count: number;
};

export const HOT_DEALS: HotDeal[] = [
  { name: 'all', count: 2 },
  { name: 'beauty', count: 2 },
  { name: 'fragrances', count: 48 },
  { name: 'furniture', count: 95 },
  { name: 'groceries', count: 23 },
];

export const BRANDS = [
  { name: 'Nike', count: 99 },
  { name: 'Adidas', count: 99 },
  { name: 'Siemens', count: 99 },
];

export const FILTER_GROUP = [{ title: 'Hot Deals', items: HOT_DEALS }];
