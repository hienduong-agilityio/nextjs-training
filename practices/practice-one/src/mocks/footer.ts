import { ROUTE } from '@/constants';

export const FOOTER_NAVIGATION = [
  {
    title: 'Information',
    items: [
      { name: 'About Us', url: ROUTE.ABOUT_US },
      { name: 'Information', url: ROUTE.INFORMATION },
      { name: 'Privacy Policy', url: ROUTE.PRIVACY_POLICY },
      { name: 'Terms & Conditions', url: ROUTE.TERM_AND_CONDITIONS },
    ],
  },
  {
    title: 'Service',
    items: [
      { name: 'About Us', url: ROUTE.ABOUT_US },
      { name: 'Information', url: ROUTE.INFORMATION },
      { name: 'Privacy Policy', url: ROUTE.PRIVACY_POLICY },
      { name: 'Terms & Conditions', url: ROUTE.TERM_AND_CONDITIONS },
    ],
  },
  {
    title: 'My Account',
    items: [
      { name: 'About Us', url: ROUTE.ABOUT_US },
      { name: 'Information', url: ROUTE.INFORMATION },
      { name: 'Privacy Policy', url: ROUTE.PRIVACY_POLICY },
      { name: 'Terms & Conditions', url: ROUTE.TERM_AND_CONDITIONS },
    ],
  },
  {
    title: 'Our Offers',
    items: [
      { name: 'About Us', url: ROUTE.ABOUT_US },
      { name: 'Information', url: ROUTE.INFORMATION },
      { name: 'Privacy Policy', url: ROUTE.PRIVACY_POLICY },
      { name: 'Terms & Conditions', url: ROUTE.TERM_AND_CONDITIONS },
    ],
  },
];
