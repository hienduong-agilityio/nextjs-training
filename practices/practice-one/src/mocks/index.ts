export * from './product';
export * from './service';
export * from './footer';
export * from './collections';
