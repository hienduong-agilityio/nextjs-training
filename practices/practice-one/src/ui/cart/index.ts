export * from './CartTable';
export * from './OrderSummary';
export * from './VoucherInput';
export * from './EmptyCart';

export { default as CartItemRow } from './CartItemRow';
