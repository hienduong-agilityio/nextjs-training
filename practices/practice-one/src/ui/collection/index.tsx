export * from './LoadingCategoryGroup';
export * from './LoadingFilterSortBar';
