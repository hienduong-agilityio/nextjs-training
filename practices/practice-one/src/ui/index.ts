export * from './products/index';
export * from './collection/index';
