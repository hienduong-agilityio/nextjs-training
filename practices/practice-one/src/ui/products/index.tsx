export * from './CartAndFavoriteActions';
export * from './ProductImages';
export * from './ProductInfo';
export { default as QuantityControl } from './QuantityControl';
export * from './SocialShare';
export * from './RelatedProducts';
export * from './ProductDetailTabs';
export * from './LoadingRelatedProducts';
export * from './LoadingProductCard';
export * from './LoadingProductTabs';
export * from './ProductReview';
