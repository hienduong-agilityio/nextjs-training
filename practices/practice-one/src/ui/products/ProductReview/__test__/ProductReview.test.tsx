import { render, screen } from '@testing-library/react';

// UI
import { ProductReviews } from '@/ui';

// Types
import type { IProductReview } from '@/interfaces';

// Mocks
import { PRODUCT_REVIEW } from '@/mocks';

describe('ProductReviews Component', () => {
  const renderProductReviews = (reviews: IProductReview[] = []) => {
    return render(<ProductReviews reviews={reviews} />);
  };

  it('matches the snapshot when there are no reviews', () => {
    const { asFragment } = renderProductReviews();
    expect(asFragment()).toMatchSnapshot();
  });

  it('renders a message when there are no reviews', () => {
    renderProductReviews();

    expect(
      screen.getByText('No reviews yet. Be the first to write one!'),
    ).toBeInTheDocument();
  });

  it('renders all reviews correctly', () => {
    renderProductReviews(PRODUCT_REVIEW);

    const reviewsElements = screen.getAllByRole('heading', { level: 4 });
    expect(reviewsElements.length).toBe(PRODUCT_REVIEW.length);

    PRODUCT_REVIEW.forEach((review) => {
      expect(screen.getByText(review.reviewerName)).toBeInTheDocument();

      expect(screen.getByText(review.reviewerEmail)).toBeInTheDocument();

      expect(screen.getByText(review.comment)).toBeInTheDocument();

      const formattedDate = new Date(review.date).toLocaleDateString();
      expect(screen.getByText(formattedDate)).toBeInTheDocument();
    });
  });
});
