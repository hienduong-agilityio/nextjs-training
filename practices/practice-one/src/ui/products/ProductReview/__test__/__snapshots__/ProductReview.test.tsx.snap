// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductReviews Component matches the snapshot when there are no reviews 1`] = `
<DocumentFragment>
  <p
    class="text-center text-gray-600 text-base md:text-lg"
  >
    No reviews yet. Be the first to write one!
  </p>
</DocumentFragment>
`;
